esp-box-3/include/bsp/esp-box-3.h

```diff
@@ -0,0 +1,463 @@
+ * by calling bsp_display_lock() before calling and LVGL API (lv_...) and then give the mutex with
+ * bsp_display_unlock().
+ *
+ * Display's backlight must be enabled explicitly by calling bsp_display_backlight_on()
+ **************************************************************************************************/
+#define BSP_LCD_H_RES              (320)
+#define BSP_LCD_V_RES              (240)
+#define BSP_LCD_PIXEL_CLOCK_HZ     (40 * 1000 * 1000)
+#define BSP_LCD_SPI_NUM            (SPI3_HOST)
+
+/**
+ * @brief Initialize display
+ *
+ * This function initializes SPI, display controller and starts LVGL handling task.
+ * LCD backlight must be enabled separately by calling bsp_display_brightness_set()
+ *
+ * @return Pointer to LVGL display or NULL when error occured
+ */
+lv_disp_t *bsp_display_start(void);
+
+/**
+ * @brief Initialize display
+ *
+ * This function initializes SPI, display controller and starts LVGL handling task.
+ * LCD backlight must be enabled separately by calling bsp_display_brightness_set()
+ *
+ * @param cfg display configuration
+ *
+ * @return Pointer to LVGL display or NULL when error occured
+ */
+lv_disp_t *bsp_display_start_with_config(const bsp_display_cfg_t *cfg);
+
+/**
+ * @brief Get pointer to input device (touch, buttons, ...)
+ *
+ * @note The LVGL input device is initialized in bsp_display_start() function.
+ *
+ * @return Pointer to LVGL input device or NULL when not initialized
+ */
+lv_indev_t *bsp_display_get_input_dev(void);
+
+/**
+ * @brief Take LVGL mutex
+ *
+ * @param timeout_ms Timeout in [ms]. 0 will block indefinitely.
+ * @return true  Mutex was taken
+ * @return false Mutex was NOT taken
+ */
+bool bsp_display_lock(uint32_t timeout_ms);
+
+/**
+ * @brief Give LVGL mutex
+ *
+ */
+void bsp_display_unlock(void);
+
+/**
+ * @brief Set display's brightness
+ *
+ * Brightness is controlled with PWM signal to a pin controling backlight.
+ *
+ * @param[in] brightness_percent Brightness in [%]
+ * @return
+ *      - ESP_OK                On success
+ *      - ESP_ERR_INVALID_ARG   Parameter error
+ */
+esp_err_t bsp_display_brightness_set(int brightness_percent);
+
+/**
+ * @brief Turn on display backlight
+ *
+ * Display must be already initialized by calling bsp_display_start()
+ *
+ * @return
+ *      - ESP_OK                On success
+ *      - ESP_ERR_INVALID_ARG   Parameter error
+ */
+esp_err_t bsp_display_backlight_on(void);
+
+/**
+ * @brief Turn off display backlight
+ *
+ * Display must be already initialized by calling bsp_display_start()
+ *
+ * @return
+ *      - ESP_OK                On success
+ *      - ESP_ERR_INVALID_ARG   Parameter error
+ */
+esp_err_t bsp_display_backlight_off(void);
+
+/**
+ * @brief Set LCD enter sleep mode
+ *
+ * Display must be already initialized by calling bsp_display_start()
+ *
+ * @return
+ *      - ESP_OK on success
+ *      - ESP_ERR_NOT_SUPPORTED if this function is not supported by the panel
+ */
+esp_err_t bsp_display_enter_sleep(void);
```

Comment:
What about make only API like this: `esp_err_t bsp_enter_sleep(void); / esp_err_t bsp_exit_sleep(void);` and there sleep everything, what was initialized?

Reply:
> What about make only API like this: `esp_err_t bsp_enter_sleep(void); / esp_err_t bsp_exit_sleep(void);` and there sleep everything, what was initialized?

Do you mean to merge LCD and touch sleep function into one API?
I think it's a good suggestion. Considering that you were not very receptive to this kind of composite function before, I split it up.
So I make an API here to turn off LCD and TP together? 
`what was initialized?`, sleep mode function will create when register touch and LCD.

Reply:
Yes, thank you. When I am thinking, the best name of the functions will be `bsp_display_enter_sleep()` and it will be control sleep for display (+brightness) and touch. 

@tore-espressif @igrr Please, what do you thing of the naming and using sleep functions in BSP?

Reply:
I see, that you added mentioned functions, but I thought left only this `bsp_display_enter_sleep()` and `bsp_display_exit_sleep()`. Other sleep functions should be deleted from public API.

Reply:
> I see, that you added mentioned functions, but I thought left only this `bsp_display_enter_sleep()` and `bsp_display_exit_sleep()`. Other sleep functions should be deleted from public API.

Hello, `bsp_display_backlight_off/on`, this API need to be retained?

Reply:
> Hello, `bsp_display_backlight_off/on`, this API need to be retained?

It should be retained. These functions are in all BSPs with display and for consistency should be there.

Reply:
> > Hello, `bsp_display_backlight_off/on`, this API need to be retained?
> 
> It should be retained. These functions are in all BSPs with display and for consistency should be there.

Done now. PTAL, thanks.